libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -276,7 +276,7 @@ void AP_ADSB::update(void)
         }
         out_state.cfg.ICAO_id_param_prev = out_state.cfg.ICAO_id_param;
         set_callsign("PING", true);
-        gcs().send_text(MAV_SEVERITY_INFO, "ADSB: Using ICAO_id %d and Callsign %s", out_state.cfg.ICAO_id, out_state.cfg.callsign);
+        gcs().send_text(MAV_SEVERITY_INFO, "ADSB: Using ICAO_id %d and Callsign %s", (int)out_state.cfg.ICAO_id, out_state.cfg.callsign);
```

Comment:
shouldn't we change it to %u and use unsigned(out_state.cfg.ICAO_id) ?
Otherwise we are limiting the range of values that can be displayed.

Reply:
@tridge, I checked them pretty thoroughly to ensure the variable type matched what was being printed.  So in this case, the "ICAO_id" is actually an int32_t (https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_ADSB/AP_ADSB.h#L163).  It's possible it should be a uint32 or a uint16 but I think that's out of scope.